core/src/main/scala/kanaloa/reactive/dispatcher/queue/Worker.scala

```diff
@@ -160,23 +160,15 @@ trait Worker extends Actor with ActorLogging with MessageScheduler {
   }
 
```

Comment:
in `retryOrAbandon`, there is a check:
```scala
if (outstanding.retried < outstanding.work.settings.retry && delayBeforeNextWork.isEmpty) {
```
This function is only called onFailure, and as such, since `delayBeforeNextWork` is only set on timeout(and no retries are ever attempted in a timeout case), this check is not needed.